Lib/test/test_calendar.py

```diff
@@ -542,25 +544,86 @@ def test_months(self):
             self.assertEqual(value[::-1], list(reversed(value)))
 
     def test_locale_calendars(self):
+        try:
+            cal = calendar.LocaleTextCalendar(locale='')
+            local_weekday = cal.formatweekday(1, 10)
+            local_weekday_abbr = cal.formatweekday(1, 3)
+            local_month = cal.formatmonthname(2010, 10, 10)
+        except locale.Error:
+            # cannot set the system default locale -- skip rest of test
+            raise unittest.SkipTest('cannot set the system default locale')
+        self.assertIsInstance(local_weekday, str)
+        self.assertIsInstance(local_weekday_abbr, str)
+        self.assertIsInstance(local_month, str)
+        self.assertEqual(len(local_weekday), 10)
+        self.assertEqual(len(local_weekday_abbr), 3)
+        self.assertGreaterEqual(len(local_month), 10)
```

Comment:
Personally, rather than one long test method that covers many cases (multiple blocks of setups and asserts), I prefer to keep each method short: test one thing.

That way, if a middle case fails, the remainder of the cases still run.

(If necessary, create a helper to make the skip bit easier.)

Reply:
Agreed. However, this particular test case takes advantage of switching between locales/another case.
So the design is correct. However, as the code gets a bit larger, we can split the cases of `LocaleTextCalendar` and `LocaleHTMLCalendar` (e73fc9e).